projects/core/tokens/icon-resolver.ts

```diff
@@ -21,12 +35,48 @@ export const TUI_ICON_RESOLVER = new InjectionToken<TuiStringHandler<string>>(
  */
 export const TUI_ICON_START_RESOLVER = TUI_ICON_RESOLVER;
 
-export function tuiInjectIconResolver(): TuiStringHandler<string> {
+export function tuiGetIconMode(icon?: string | null): TUI_ICON_MODE | null {
+    if (!icon) {
+        return null;
+    }
+
+    if (icon.startsWith(ICON_MODE_PREFIXES.image)) {
+        return 'image';
+    }
+
+    if (icon.startsWith(ICON_MODE_PREFIXES.font)) {
+        return 'font';
+    }
+
+    return 'svg';
+}
+
+export function tuiInjectIconResolver(
+    tag: ICON_TARGET_TAG = 'url',
```

Comment:
There shouldn't be any modes for injecting the resolver, it would be a breaking change if it starts adding `url(`. Instead if the name starts with `'@font.` it must remove the prefix and return the rest, otherwise it should behave the same way it behaved before. And in every place where it is used we need to take care of adding `url(` or not adding it, if it's a font.